modules/Hoot/ui/managePanel/transAssist/tagMapWidget.js

```diff
@@ -29,20 +29,38 @@ export default class TagMapWidget {
             .append( 'div' )
             .classed( 'pad1 thumbnail searchtag _icon big blank search-icon keyline-right', true );
 
-        this.searchTag = this.inputWrapper.append( 'input' )
+        this.searchTag = this.inputWrapper.append('input')
             .attr( 'type', 'text' )
             .attr( 'placeholder', 'Search Tag' )
             .classed( 'strong bigger pad1x pad2y reset', true )
-            .on( 'input', () => this.change.call( this ) )
+            .on( 'input', () => this.change.call(this) )
             .on( 'keyup', () => that.keyup() )
-            .on( 'keydown', () => that.keydown( this ) );
+            .on( 'keydown', () => that.keydown(this) );
 
 
         this.resultsList = this.tagLookup
             .append( 'div' )
             .classed( 'results-list', true );
 
         this.searchTag.node().focus();
+
+        if (!Hoot.config.tagInfo[this.schemaOption]) {
+            Hoot.config.tagInfo[this.schemaOption] = [];
+        }
+        if (Hoot.config.tagInfo[this.schemaOption].length === 0) {
+            (this.schemaOption === 'OSM'
+                ?  new Promise(function(resolve, reject) {
+                    services.taginfo.keys({}, function(err, data) {
+                        if (!err) {resolve(data.map(v => { return v.value; }));}
+                        else { reject(err);}
+                    });
```

Comment:
I think this call needs to move to the `change()` function so that it can pass a query param.  For example, I type the tag `amenity` but it must not be in the top 10 most popular tag keys.  When doing this the manual filtering in `change()` won't be needed for taginfo results.

Reply:
with changes in [1696e58](https://github.com/ngageoint/hootenanny-ui/pull/2032/commits/1696e58ac0c5af87c3164a7a98480f16ab30d549) think we also want the if block on line 51 to be `is tagInfo list empty and is schema not osm` then it should only invoke the `getFieldMappings` func.